src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs

```diff
@@ -1787,6 +1787,19 @@ public void WriteMetadataAndIL(PdbWriter nativePdbWriterOpt, Stream metadataStre
 
             if (_debugMetadataOpt != null)
             {
+                // Ensure document table lists files in command line order
+                // This is key for us to be able to accurately rebuild a binary from a PDB.
+                var documentsBuilder = Module.DebugDocumentsBuilder;
+                var documents = documentsBuilder.DebugDocuments;
+                foreach (var tree in Module.CommonCompilation.SyntaxTrees)
+                {
+                    string normalizedPath = documentsBuilder.NormalizeDebugDocumentPath(tree.FilePath, basePath: null);
+                    if (documents.TryGetValue(normalizedPath, out var doc) && !_documentIndex.ContainsKey(doc))
```

Comment:
`if (documents.TryGetDebugDocument(tree.FilePath, basePath: null) is { } doc && ...`

Reply:
Fixed